lua/sniprun.lua

```diff
@@ -84,11 +94,26 @@ function M.setup(opts)
   M.config_up = 1
 end
 
+
+local highlight = function(group, styles)
```

Comment:
I see an issue here.

I'd like the highlight to be defined either by: the default, an external colorscheme, a custom config

The override should follow this priority: custom config > external colorscheme > default.

Reply:
Yeah . There does seem to be an issue with default highlighting

But the look up is still (config_file -> colorscheme -> default)
My Sniprun configuration file has 
```lua 
require'sniprun'.setup({
  SniprunColors = {
    SniprunVirtualTextOk   =  {bg="#00ff00",fg="#000000",gui='italic'},
    SniprunFloatingWinOk   =  {bg="#00ff00",fg="#000000",gui='underline'},
    SniprunVirtualTextErr  =  {bg="#ff0000",fg="#000000",gui='bold'},
    SniprunFloatingWinErr  =  {bg="#ff0000",fg="#000000"}
  }
})
```
and my colorscheme has
```lua

  SniprunVirtualTextOk = {bg="#a3be8c",fg="#1e222a",style='italic'},
  SniprunVirtualTextErr = {bg="#d08770",fg="#1e222a",style='bold'},
  SniprunFloatingWinErr = {bg="#d08770",fg="#1e222a",style='bold'},
  SniprunFloatingWinOk = {bg="#a3be8c",fg="#1e222a",style='italic'},

```

with both Sourced . This is the result

![hl](https://user-images.githubusercontent.com/46932291/116047976-4ea7fc00-a692-11eb-9126-b799241715a1.png)

Which it clearly got it from the configuration file

In the absence of a configuration file it looks in my colorscheme 

![from_colorscheme](https://user-images.githubusercontent.com/46932291/116048485-dbeb5080-a692-11eb-9bfb-f12137fe1a30.png)

And if the highlight group is missing from both . 
Dont really know why it's behaving this way

![2021 04 26-14 03 29 screenshot](https://user-images.githubusercontent.com/46932291/116053689-96ca1d00-a698-11eb-9275-8ebd21e32099.png)

But after reloading the colorscheme the default colors kicks in.


![2021 04 26-14 04 04 screenshot](https://user-images.githubusercontent.com/46932291/116054123-fd4f3b00-a698-11eb-83e7-f075d0ca6683.png)

Reply:
Alright, that's a satisfying behavior :-) ("reloading" the colorscheme might be an akward way to put it, though, even if I can understand it means that removes the sniprun highlights from the colorscheme)

Concerning the virtual text being displayed there, it's probably because your selection ended on that last empty line. Finer-tuned behavior is implemented, but not finished, and not on master yet. You don't have to concern yourself with this. 

Amazing job otherwise !